src/codeql.ts

```diff
@@ -131,3 +110,93 @@ async function downloadDatabase(
     }
   }
 }
+
+// Calls `bqrs info` and returns compatible query kinds for the given bqrs file
+async function getCompatibleQueryKinds(
+  codeql: string,
+  bqrs: string
+): Promise<string[]> {
+  const bqrsInfoOutput = await getExecOutput(codeql, [
+    "bqrs",
+    "info",
+    "--format=json",
+    bqrs,
+  ]);
+  if (bqrsInfoOutput.exitCode !== 0) {
+    throw new Error(
+      `Unable to run codeql bqrs info. Exit code: ${bqrsInfoOutput.exitCode}`
+    );
+  }
+  return JSON.parse(bqrsInfoOutput.stdout)["compatible-query-kinds"];
+}
+
+// Generates results.csv from the given bqrs file
+async function outputCsv(codeql: string, bqrs: string): Promise<string> {
+  const csv = path.join("results", "results.csv");
+  await exec(codeql, [
+    "bqrs",
+    "decode",
+    "--format=csv",
+    `--output=${csv}`,
+    bqrs,
+  ]);
+  return csv;
+}
+
+// Generates results.md from the given bqrs file
+async function outputMd(
+  codeql: string,
+  bqrs: string,
+  nwo: string,
+  compatibleQueryKinds: string[]
+): Promise<string> {
+  const json = path.join("results", "results.json");
+  await exec(codeql, [
+    "bqrs",
+    "decode",
+    "--format=json",
+    `--output=${json}`,
+    "--entities=all",
+    bqrs,
+  ]);
+
+  const sourceLocationPrefix = JSON.parse(
+    (await getExecOutput(codeql, ["resolve", "database", "db"])).stdout
+  ).sourceLocationPrefix;
+
+  // This will load the whole result set into memory. Given that we just ran a
+  // query, we probably have quite a lot of memory available. However, at some
+  // point this is likely to break down. We could then look at using a streaming
+  // parser such as http://oboejs.com/
+  const jsonResults = JSON.parse(fs.readFileSync(json, "utf8"));
+
+  const md = path.join("results", "results.md");
+  const s = fs.createWriteStream(md, {
+    encoding: "utf8",
+  });
+
+  await interpret(
+    s,
+    jsonResults,
+    nwo,
+    compatibleQueryKinds,
+    sourceLocationPrefix,
+    "HEAD"
+  );
+  return md;
+}
+
+// Generates results.sarif from the given bqrs file
+async function outputSarif(codeql: string, bqrs: string): Promise<string> {
+  const sarif = path.join("results", "results.sarif");
+  await exec(codeql, [
+    "bqrs",
+    "interpret",
+    "--format=sarif-latest",
+    `--output=${sarif}`,
+    "-t=kind=problem",
```

Comment:
These two options were required by `codeql bqrs interpret`. I think we have to just make something up for these, because the query might not have them, and even if it did then getting them from the query text could be difficult. I don't think the values matter particularly right now. Maybe in the far future if there's some option to import the results into code scanning it could be important, but that's not a realistic possibility for now.

Reply:
Couldn't this argument's value come from the output of `codeql bqrs info`? This path should ideally work for both `problem` and `path-problem`.

Reply:
For info, here's what the `codeql bqrs info --format=json` output looks like
```
{
  "result-sets" : [
    {
      "name" : "#select",
      "rows" : 0,
      "columns" : [
        {
          "name" : "getInputCall",
          "kind" : "e"
        },
        {
          "kind" : "s"
        },
        {
          "name" : "inputName",
          "kind" : "s"
        },
        {
          "name" : "inputName",
          "kind" : "s"
        },
        {
          "name" : "action",
          "kind" : "e"
        },
        {
          "kind" : "s"
        }
      ]
    }
  ],
  "compatible-query-kinds" : [
    "Problem",
    "Table",
    "Tree",
    "Graph",
    "Diagnostic"
  ]
}
```
So the id is completely gone, and the kind is determinable from the compatible query kinds. So you're right we do want this to work for path problems too.

Reply:
:+1: A fake id should be fine AFAICT